saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/SavedSitesRepository.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.savedsites.api
+
+import com.duckduckgo.savedsites.api.models.BookmarkFolder
+import com.duckduckgo.savedsites.api.models.BookmarkFolderItem
+import com.duckduckgo.savedsites.api.models.FolderBranch
+import com.duckduckgo.savedsites.api.models.SavedSite
+import com.duckduckgo.savedsites.api.models.SavedSite.Bookmark
+import com.duckduckgo.savedsites.api.models.SavedSite.Favorite
+import com.duckduckgo.savedsites.api.models.SavedSites
+import io.reactivex.Single
+import kotlinx.coroutines.flow.Flow
+
+interface SavedSitesRepository {
+
+    suspend fun getSavedSites(folderId: String): Flow<SavedSites>
+
+    suspend fun getFolderContent(folderId: String): Flow<Pair<List<Bookmark>, List<BookmarkFolder>>>
+
+    suspend fun getFolderTree(
+        selectedFolderId: String,
+        currentFolder: BookmarkFolder?,
+    ): List<BookmarkFolderItem>
+
+    suspend fun insertFolderBranch(branchToInsert: FolderBranch)
+
+    fun deleteFolderBranch(folder: BookmarkFolder): FolderBranch
+
+    fun getFolderBranch(folder: BookmarkFolder): FolderBranch
+
+    fun getBookmarks(): Flow<List<Bookmark>>
+
+    fun getBookmarksObservable(): Single<List<Bookmark>>
+
+    fun getBookmark(url: String): Bookmark?
+
+    fun getFavorites(): Flow<List<Favorite>>
+
+    fun getFavoritesObservable(): Single<List<Favorite>>
+
+    fun getFavoritesSync(): List<Favorite>
```

Comment:
is this something we forgot to remove? not sure we want to see references to sync inside this module.